addons/purchase_stock/models/purchase_order.py

```diff
@@ -251,9 +251,8 @@ def _get_picking_type(self, company_id):
         picking_type = self.env['stock.picking.type'].search([('code', '=', 'incoming'), ('warehouse_id.company_id', '=', company_id)])
         if not picking_type:
             picking_type = self.env['stock.picking.type'].search([('code', '=', 'incoming'), ('warehouse_id', '=', False)])
-        company_warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
-        if not company_warehouse:
-            self.env['stock.warehouse']._warehouse_redirect_warning()
+        if not picking_type:
+            picking_type = self.env['stock.picking.type'].with_context(active_test=False).search([('code', '=', 'incoming'), ('warehouse_id', '=', False)])
```

Comment:
```suggestion
        if not picking_type:
            picking_type = self.env['stock.picking.type'].with_context(active_test=False).search([('code', '=', 'incoming'), ('warehouse_id', '=', False)])
        company_warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
        if not picking_type and not company_warehouse:
            self.env['stock.warehouse']._warehouse_redirect_warning()
```
Could this work? So we include a bypass if there's an archived picking type, but it still leaves the redirect for everyone else?

Reply:
@clesgow sorry for the late reply. I checked, and your suggested patch did not work because the warehouse is inactive, which caused the redirect to be triggered.

Reply:
@clesgow The main issue occurs when the related picking type is archived and the also warehouse is inactive, which triggers a redirect. I had previously opened a [PR](https://github.com/odoo/upgrade/pull/7037) in the upgrade repository to address this, but it was closed due to the opening of this current PR.

Additionally, I've encountered this archived picking type scenario, and the warehouse is inactive in multiple upgrade requests. So far, 17 upgrade requests have failed due to the following commit:
https://github.com/odoo/odoo/commit/6516ab61927a63e3f2d804cf1b5baa43a151ca19

Please let me know if you need any further information.

Reply:
@bda-odoo But in the suggested code above, you'd have the redirect warning only if there isn't any archived picking type scenario *and* no warehouse? So as long as you have an archived picking type, you're good to go, aren't you?

Reply:
@bda-odoo So how did that not work when you tried it? I'm very confused.

Reply:
@clesgow, [This patch](https://github.com/odoo/odoo/pull/194897/files#diff-96229de0cd95cec1b8dd72005b6717a1bb6d8f0066057b68770e9dc8d2a70618R254-R255) is working fine because it does not involve checking the company warehouse.

Please let me know if you need any further information.